app/models/notification.rb

```diff
@@ -0,0 +1,21 @@
+class Notification < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+  belongs_to :user
+  belongs_to :course
+
+  scope :center_popup, -> { where(type: 'CenterPopup') }
+  scope :right_side_popup, -> { where(type: 'RightSidePopup') }
+  scope :email, -> { where(type: 'EmailNotification') }
+
+  def self.types
+    %w(CenterPopup RightSidePopup EmailNotification)
+  end
+
+  def self.set_options(notification, options = {})
```

Comment:
Can you properly document this methods ? I still don't know how to use it after looking at your code for a few times..